linetools/spectralline.py

```diff
@@ -0,0 +1,370 @@
+
+    # Output
+    def __repr__(self):
+        txt = '[{:s}:'.format(self.__class__.__name__)
+        try:
+            txt = txt+' {:s},'.format(self.data['name'])
+        except KeyError:
+            pass
+        txt = txt + ' wrest={:g}'.format(self.wrest)
+        txt = txt + ']'
+        return (txt)
+
+# Class for Generic Absorption Line System
+class AbsLine(SpectralLine):
+    """Spectral absorption line
+    """
+    # Initialize with a .dat file
+    def __init__(self, wrest, linelist=None): 
+        # Generate with type
+        SpectralLine.__init__(self,'Abs', wrest, linelist=linelist)
+
+    def print_specline_type(self):
+        """"Return a string representing the type of vehicle this is."""
+        return 'AbsLine'
+
+    def fill_data(self,linelist=None):
+        ''' Fill atomic data and setup analy
+        Parameters:
+        -----------
+        linelist : LineList, optional
+          Class of linelist or str setting LineList
+        '''
+
+        # Deal with LineList
+        if linelist is None:
+            self.llist = LineList('ISM')
+        elif isinstance(linelist,basestring):
+            self.llist = LineList(linelist)
+        elif isinstance(linelist,LineList):
+            self.llist = linelist
+        else:
+            raise ValueError('Bad input for linelist')
+
+        # Data
+        self.data.update(self.llist[self.wrest])
+
+        #
+        self.analy.update( {
+            'FLG_EYE': 0,
+            'FLG_LIMIT': 0, # No limit
+            'DATFIL': '', 
+            'IONNM': self.data['name']
+            })
+
+        # Additional attributes for Absorption Line
+        self.attrib.update({'N': 0., 'Nsig': 0., 'flgN': 0, # Column
+                       'b': 0., 'bsig': 0.  # Doppler
+                       } )
+
+    # Perform AODM on the line
+    def aodm(self, **kwargs): 
+        """  AODM calculation
+
+        Parameters
+        ----------
+        spec : Spectrum1D (None)
+          1D spectrum.  Required but often read in through the Class (self.spec)
+        conti : np.array (None)
+          Continuum array 
+
+        Returns:
+          N, sigN : Column and error in linear space
+        """
+
+        # Grab Spectrum
+        spec = self.set_spec(**kwargs)
+
+        # Velocity array
+        spec.velo = spec.relative_vel(self.wrest*(1+self.analy['z']))
+
+        # Pixels for evaluation
+        pix = spec.pix_minmax(self.analy['z'], self.wrest,
+                        self.analy['VLIM'].to('km/s'))[0]
+                        #self.analy['VLIM'].to('km/s').value)[0]
+
+        # For convenience + normalize
+        velo = spec.velo[pix]
+        fx, sig = parse_spec(spec, **kwargs)
+
+        # dv
+        delv = velo - np.roll(velo,1)
+        delv[0] = delv[1]
+
+        # Atomic data
+        cst = (10.**14.5761)/(self.data['fval']*self.wrest) / (u.km/u.s) / u.cm * (u.AA/u.cm)
+
+        # Mask
+        mask = (pix == pix) # True = good
```

Comment:
`pix == pix` is neat, but to me it's clearer to use `mask = np.ones(len(pix), dtype=bool)`. But your call.